test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/PushCommandTest.cs

```diff
@@ -245,6 +256,512 @@ public void PushCommand_Server_SkipDuplicate_IsSpecified_PushProceeds()
+        {
+            // Arrange
+            using (var packageDirectory = TestDirectory.Create())
+            {
+                var nuget = Util.GetNuGetExePath();
+
+                string packageId = "packageWithSnupkg";
+
+                //Create a nupkg in test directory.
+                string version = "1.1.0";
+                Util.CreateTestPackage(packageId, version, packageDirectory);
+                string nupkgFileName = Util.BuildPackageString(packageId, version, NuGetConstants.PackageExtension);
+                string snupkgFileName = Util.BuildPackageString(packageId, version, NuGetConstants.SnupkgExtension);
+                string snupkgFullPath = Path.Combine(packageDirectory, snupkgFileName);
+                //Create snupkg in test directory.
+                WriteSnupkgFile(snupkgFullPath);
+
+                string wildcardPush = "*.nupkg";
+
+                using (var server = CreateAndStartMockV3Server(packageDirectory, out string sourceName))
+                {
+                    //Configure push to return a Conflict for the first push, then Created for all remaining pushes.
+                    SetupMockServerCreateNupkgDuplicateSnupkg(server, packageDirectory, FuncStatus_Duplicate_ThenAlwaysCreated());
+
+                    // Act
+
+                    //Since this is V3, this will trigger 2 pushes: one for nupkgs, and one for snupkgs.
+                    var result = CommandRunner.Run(
+                        nuget,
+                        packageDirectory,
+                        $"push {wildcardPush} -Source {sourceName} -Timeout 110",
+                        waitForExit: true,
+                        timeOutInMilliseconds: 120000); // 120 seconds
+
+                    // Assert
+
+                    //Ignoring filename in File Not Found error since the error should not appear in any case.
+                    string genericFileNotFoundError = string.Format(MESSAGE_FILE_DOES_NOT_EXIST, string.Empty);
+
+                    //Nupkg should be a conflict, so its snupkg should also not push.
+                    Assert.False(result.Success, "Expected to fail the push due to a duplicate nupkg.");
+                    Assert.DoesNotContain(MESSAGE_PACKAGE_PUSHED, result.Output); //nothing pushed
+                    Assert.Contains(MESSAGE_RESPONSE_NO_SUCCESS, result.Errors); //nupkg duplicate
+                    Assert.DoesNotContain(genericFileNotFoundError, result.Errors);
+                    Assert.DoesNotContain(".snupkg", result.AllOutput); //snupkg not mentioned
+                }
+            }
+        }
+
+        /// <summary>
+        /// When pushing *.Nupkg with SkipDuplicate, a 409 Conflict is ignored and the secondary symbols push proceeds.
+        /// TODO: bug fixes will come from https://github.com/NuGet/Home/issues/8148
+        /// </summary>
+        [Fact]
+        public void PushCommand_Server_Nupkg_WildcardFindsMatchingSnupkgs_SkipDuplicate()
```

Comment:
The following tests looks similar:

- `PushCommand_Server_Nupkg_WildcardFindsMatchingSnupkgs_SkipDuplicate`
- `PushCommand_Server_Snupkg_WildcardFindsMatchingSnupkgs_SkipDuplicate`

Can those tests generalized using a `[Theory]`? Does the behavior will be different once NuGet/HOme#8148 is solved?

The same comment also applies for all the tests.

Reply:
They are similar, but very different code paths occur when pushing Nupkgs vs. Snupkgs, directly.